irohad/ametsuchi/impl/rocksdb_wsv_command.cpp

```diff
@@ -0,0 +1,472 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/rocksdb_wsv_command.hpp"
+
+#include <numeric>
+
+#include <fmt/core.h>
+#include <boost/format.hpp>
+#include "ametsuchi/impl/executor_common.hpp"
+#include "ametsuchi/impl/rocksdb_common.hpp"
+#include "ametsuchi/ledger_state.hpp"
+#include "backend/protobuf/permissions.hpp"
+#include "interfaces/common_objects/account.hpp"
+#include "interfaces/common_objects/account_asset.hpp"
+#include "interfaces/common_objects/asset.hpp"
+#include "interfaces/common_objects/domain.hpp"
+#include "interfaces/common_objects/peer.hpp"
+
+namespace iroha::ametsuchi {
+
+  template <typename Func, typename Error>
+  WsvCommandResult execute(std::shared_ptr<RocksDBContext> &context,
+                           Func &&func,
+                           Error &&error) {
+    RocksDbCommon common(context);
+    if (auto result = std::forward<Func>(func)(common);
+        expected::hasError(result))
+      return expected::makeError(
+          fmt::format("Command: {}, DB error: {} with description {}",
+                      std::forward<Error>(error)(),
+                      result.assumeError().code,
+                      result.assumeError().description));
+
+    common.commit();
```

Comment:
```suggestion
```
Are you sure that we would need to commit after each operation? I would rather have this class perform state changes, but leave transaction management outside it.